enable Dependabot v2

sullis opened the pull request.

sullis commented:
@mkurz

sullis commented:
@mkurz

ignasi35 commented:
Sounds like a good addition. I'm a bit skeptical about the `daily` periodicity, though. I other projects using `scalasteward` we moved to a monthly period and it proved to be frequent enough.

sullis commented:
> Sounds like a good addition. I'm a bit skeptical about the `daily` periodicity, though. I other projects using `scalasteward` we moved to a monthly period and it proved to be frequent enough.

Sounds good.    I just updated it.

ennru commented:
Dependabot is already active on this repo https://github.com/playframework/netty-reactive-streams/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+sort%3Aupdated-desc

ignasi35 commented:
> Dependabot is already active on this repo [playframework/netty-reactive-streams/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+sort%3Aupdated-desc](https://github.com/playframework/netty-reactive-streams/pulls?q=is%3Apr+author%3Aapp%2Fdependabot+sort%3Aupdated-desc)

Is it possible that it was active, but it's not anymore?
I mean, there's only one PR on the link you provided and it's updating Netty to a version (4.1.45) which is not the latest so I'm assuming dependabot is not active or maybe missconfigured because we're missing version bumps.
